org.adempiere.pipo/src/org/adempiere/pipo2/PoFiller.java

```diff
@@ -408,7 +412,33 @@ private void setBlob(String qName) {
 						throw new AdempiereException(e.getLocalizedMessage(), e);
 					}
 				}
-				po.set_ValueNoCheck(qName, data);
+				if ("BinaryData".equals(qName) && data instanceof byte[]) {
+					if (po instanceof MArchive) {
+						/* it comes as a zip file with a single PDF file */
+					    ZipInputStream zipStream = new ZipInputStream(new ByteArrayInputStream((byte[]) data));
+					    byte[] output = null;
+						try {
+							while (zipStream.getNextEntry() != null) {

```

Comment:
the while loop here looks strange since each loop will just replace what's in the output byte array variable.